### PR TITLE
quic: additional edits and multiple endpoint start

### PR DESCRIPTION
**Pull request by jasnell**
Additional edits + Beginning support for multiple local UDP endpoints

### CODE REVIEW

File: src/node_sockaddr.h [hunk 1]
```diff
@@ -72,18 +72,50 @@ class SocketAddress {
       uv_tcp_t* handle,
       SocketAddress* addr = nullptr);
```
Comment (addaleax) [inline src/node_sockaddr.h]: The four overloads taking a non-const pointer are no longer necessary, right?

Comment (jasnell) [reply src/node_sockaddr.h]: Yeah, I'll be pulling those back out. This is still very much a WIP

**Comment by jasnell** [comment]
Will get this PR updated tomorrow

**Comment by jasnell** [comment]
There's still a lot to do to get QuicSocket's ready for multiple udp endpoints but this lays the ground work. Will do the rest in a separate PR

File: src/udp_wrap.h [hunk 2]
```diff
@@ -46,7 +46,8 @@ class UDPListener {
   // Called right after data is received from the socket, and includes
   // information about the source address. If `nread` is negative, an error
   // has occurred, and it represents a libuv error code.
-  virtual void OnRecv(ssize_t nread,
+  virtual void OnRecv(uv_udp_t* handle,
```
Comment (addaleax) [inline src/udp_wrap.h]: This doesn’t seem like something we should be doing … currently, `UDPListener` is already associated with one `UDPWrapBase` instance, so this should always be same same value and always retrievable

As far as I can tell, this is currently only being used for the `SocketAddress::FromSockName()` call above? I might be missing something, but that one could probably use `udp_->GetSockName()` instead?

Comment (jasnell) [reply src/udp_wrap.h]: Ok, will rework it!

**Comment by jasnell** [comment]
Landed

**Pull request closed without merging**